### PR TITLE
Use blank line instead of `@desc:` for doc comments

### LINKED ISSUE #3400: Remove `@desc` tag in documentation comments
### Describe the project you are working on

A game

### Describe the problem or limitation you are having in your project

[Documentation comments](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_documentation_comments.html#tags) added in 4.0-dev. The brief does not require a tag, but the description does, although the description is usually longer than the brief.

In this case, for the description, you need to add 4 spaces at the beginning of each line. This means less width space (quite a few people have narrow screens or try to stick to the 80-100 characters per line limit). Plus a slightly larger file size (not that important, but why increase the file size if it doesn't make sense?).

```gdscript
##
## A brief description of your script.
##
## @desc:
##     A more detailed description of the script.[br]
##     Lorem ipsum dolor sit amet, consectetur adipiscing elit,
##     sed do eiusmod tempor incididunt ut labore et dolore
##     magna aliqua.[br]
##     Ut enim ad minim veniam, quis nostrud exercitation ullamco
##     laboris nisi ut aliquip ex ea commodo consequat.[br]
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
##
```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The first option is to swap the brief and description, i.e. require a tag for the first and not require a tag for the second:

<details>
<summary>Option 1.1</summary>

```gdscript
##
## @brief:
##     A brief description of your script.
##
## A more detailed description of the script.[br]
## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
## do eiusmod tempor incididunt ut labore et dolore magna aliqua.[br]
## Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
## nisi ut aliquip ex ea commodo consequat.
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
##
```

</details>

We should also allow the brief to be on the same line as the tag:

<details>
<summary>Option 1.2</summary>

```gdscript
##
## @brief: A brief description of your script.
##
## A more detailed description of the script.[br]
## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
## do eiusmod tempor incididunt ut labore et dolore magna aliqua.[br]
## Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
## nisi ut aliquip ex ea commodo consequat.
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
##
```

</details>

The first option has 2 drawbacks: some may not want to use descriptions at all, but they will have to use `@brief` every time; Also in doc comments, `@tags` are usually placed at the end. We can, by analogy with commit messages in git, split a comment into a brief and a description by an empty line between them:

<details>
<summary>Option 2</summary>

```gdscript
##
## A brief description of your script.
##
## A more detailed description of the script.[br]
## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
## do eiusmod tempor incididunt ut labore et dolore magna aliqua.[br]
## Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
## nisi ut aliquip ex ea commodo consequat.[br]
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
##
```

</details>

Just a variation of Option 2:

<details>
<summary>Option 3</summary>

```gdscript
##
## A brief description of your script.
##
##
## A more detailed description of the script.
##
## Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed
## do eiusmod tempor incididunt ut labore et dolore magna aliqua.
##
## Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
## nisi ut aliquip ex ea commodo consequat.
##
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
##
```

</details>

By the way, doc comments do not support `[p][/p]` BBCode, although `RichTextLabel` does.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

This cannot be done with an add-on.

### PR DESCRIPTION
**Pull request by dalexeev**
Closes godotengine/godot-proposals#3400.

Now the description in the doc comment is separated from the brief with an empty line (like git commit messages).

<details>
<summary><b>Before</b></summary>

```gdscript
## A brief description of your script.
##
## @desc:
##     A more detailed description of the script.[br]
##     Lorem ipsum dolor sit amet, consectetur adipiscing elit,
##     sed do eiusmod tempor incididunt ut labore et dolore
##     magna aliqua.[br]
##     Ut enim ad minim veniam, quis nostrud exercitation ullamco
##     laboris nisi ut aliquip ex ea commodo consequat.[br]
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
```

<details>
<summary>This works too</summary>

```gdscript
## A brief description of your script.
##
## @desc:
## A more detailed description of the script.[br]
## Lorem ipsum dolor sit amet, consectetur adipiscing elit,
## sed do eiusmod tempor incididunt ut labore et dolore
## magna aliqua.[br]
## Ut enim ad minim veniam, quis nostrud exercitation ullamco
## laboris nisi ut aliquip ex ea commodo consequat.[br]
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
```

</details>

<details>
<summary>And this</summary>

```gdscript
## A brief description of your script.
##
## @desc: A more detailed description of the script.[br]
## Lorem ipsum dolor sit amet, consectetur adipiscing elit,
## sed do eiusmod tempor incididunt ut labore et dolore
## magna aliqua.[br]
## Ut enim ad minim veniam, quis nostrud exercitation ullamco
## laboris nisi ut aliquip ex ea commodo consequat.[br]
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
```

</details>

But isn't it better to do without `@desc:` at all?

</details>

<details>
<summary><b>After</b></summary>

```gdscript
## A brief description of your script.
##
## A more detailed description of the script.[br]
## Lorem ipsum dolor sit amet, consectetur adipiscing elit,
## sed do eiusmod tempor incididunt ut labore et dolore
## magna aliqua.[br]
## Ut enim ad minim veniam, quis nostrud exercitation ullamco
## laboris nisi ut aliquip ex ea commodo consequat.[br]
##
## @tutorial:            http://the/tutorial1/url.com
## @tutorial(Tutorial2): http://the/tutorial2/url.com
```

</details>

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -3135,10 +3135,9 @@ void GDScriptParser::get_class_doc_comment(int p_line, String &p_brief, String &
 		String striped_line = doc_line.strip_edges();
 
 		// Set the read mode.
-		if (striped_line.begins_with("@desc:") && p_desc == "") {
+		if (striped_line.is_empty() && mode == BRIEF) {
```
Comment (dalexeev) [inline modules/gdscript/gdscript_parser.cpp:3138]: Is the `!p_brief.is_empty()` check necessary here? Should we allow blank lines like:

```gdscript
##
## Brief.
##
## Description.
##
```

Or we will only allow:

```gdscript
## Brief.
##
## Description.
```

Comment (Calinou) [reply modules/gdscript/gdscript_parser.cpp]: I personally wouldn't allow extraneous lines at the top and bottom, but that's just me.

**Review by akien-mga — APPROVED** [review]
That looks good to me.

File: modules/gdscript/gdscript_parser.cpp [hunk 2]
```diff
@@ -3141,10 +3141,9 @@ void GDScriptParser::get_class_doc_comment(int p_line, String &p_brief, String &
 		String striped_line = doc_line.strip_edges();
 
 		// Set the read mode.
-		if (striped_line.begins_with("@desc:") && p_desc.is_empty()) {
+		if (striped_line.is_empty() && mode == BRIEF && !p_brief.is_empty()) {
```
[the remark this answers is not in the document]
Comment (dalexeev) [reply modules/gdscript/gdscript_parser.cpp]: Done.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**